### PR TITLE
Disable 100-continue conditionally

### LINKED ISSUE #131: 100-continue and servers that does not support that
Curl if used by openwsman library always sends 100-continue [1].
We have bunch of embedded servers that does not support that header, so
curl continues after 1sec timeout. That creates big delays for long conversations.

Now in our code I hard-disable the header:
```
diff -ru openwsman/src/lib/wsman-curl-client-transport.c ./src/lib/wsman-curl-client-transport.c
--- openwsman/src/lib/wsman-curl-client-transport.c        2019-09-24 10:59:52.634048988 +0300
+++ ./src/lib/wsman-curl-client-transport.c     2019-09-18 17:08:50.494829963 +0300
@@ -521,9 +521,6 @@
        }
 #endif

-       /* Don't request 100-continue*/
-       headers = curl_slist_append(headers, "Expect:");
-
        r = curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
        if (r != CURLE_OK) {
                cl->fault_string = u_strdup(curl_easy_strerror(r));
```

Is it the right way to do?
Can we make it default openwsman behavior?
Or add some parameter to have it as option without patching code?

[1] https://gms.tf/when-curl-sends-100-continue.html

### PR DESCRIPTION
**Pull request by ausyskin**
Fixes #131 

The C++ API is changed in this PR, so it need an API version bump.

### CODE REVIEW

File: src/cpp/OpenWsmanClient.cpp [hunk 1]
```diff
@@ -207,9 +208,10 @@ string OpenWsmanClient::Get(
 string OpenWsmanClient::Put(
 	const string &resourceUri,
 	const string &content,
-	const NameValuePairs *s) const
+	const NameValuePairs *s,
+	unsigned long flags) const
```
Comment (kkaempf) [inline src/cpp/OpenWsmanClient.cpp]: Same as above. Would `OpenWsmanClient::PutWithFlags` be better ?

File: src/cpp/OpenWsmanClient.cpp [hunk 2]
```diff
@@ -195,9 +195,10 @@ string OpenWsmanClient::Get(
 
 string OpenWsmanClient::Get(
 	const string &resourceUri,
-	const NameValuePairs *s) const
+	const NameValuePairs *s,
+	unsigned long flags) const
```
Comment (kkaempf) [inline src/cpp/OpenWsmanClient.cpp]: Hmm, this changes the API. I'm not sure if this is the right approach or if introducing `OpenWsmanClient::GetWithFlags` would be a better one.

Comment (ausyskin) [reply src/cpp/OpenWsmanClient.cpp]: Ok, will split to the new functions

Comment (ausyskin) [reply src/cpp/OpenWsmanClient.cpp]: Done, please look if this fit.

Comment (ausyskin) [reply src/cpp/OpenWsmanClient.cpp]: Is the new split is ok?

**Comment by ausyskin** [comment]
The curl 7.69.0 pushes limit for "expect" header to 1MB so, this patch is not relevant for typical WsMan use-case. I may close this pull request, if you wish.

**Comment by kkaempf** [comment]
The changes don't harm, merging. Thanks for your patience, Alexander !

**Pull request merged**